apps/yggdrasil/lib/yggdrasil/user.ex

```diff
@@ -29,14 +33,46 @@ defmodule Yggdrasil.User do
   end
 
   @doc """
-  adds password at end of chain protects the hashpwsalt from seeing a nil value
-  in the case of missing password field.
+  Creates the user and assigns the default roles of "player"
   """
-  def hash_password(changeset = %{:valid? => false}) do
+  def create_with_default_role(attributes) do
+    changeset = User.create_changeset(%User{}, attributes)
+
+    with {:ok, user} <- Repo.insert(changeset),
+          :ok        <- add_default_role(user),
+     do: {:ok, user}
```

Comment:
This is begging for a transaction to be wrapped around it. I did actually wrap this in a transaction, and found out that it does some odd behavior with the test transaction that the sandbox adapter that is opened before each test.

It seems that if this nested transaction is committed then then so is the test transaction. I noted this when the registration_controller_test started having kittens about username already exists which violates the unique name constraint on the table. Seems to be directly tied to to the transaction I had.

I dropped it for now, but transaction probably should happen higher up anyway like in the registration controller or other controllers but obviously it would not solve the issue with the sandbox. Going to look into that when I get a chance, but for now just leaving this as is.